### PR TITLE
chore: eval selection logic much simpler

### PR DESCRIPTION
**Pull request by gtarpenning**
https://wandb.atlassian.net/browse/WB-19856

This pr: 
- makes comparison button primary
- removes bulk delete mode
- sets the max selection in trace view to 100
- removes the disabled selection of un-comparable evals
     - this was causing confusion, both Morgan and CVP didn't like this. 
- make evals not crash if there isn't an output (failed or running)
- show eval display name in comparison page

Screenshot 2024-07-29 at 1 14 36 PM

### CODE REVIEW

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTable.tsx [hunk 1]
```diff
@@ -81,7 +82,7 @@ import {ManageColumnsButton} from './ManageColumnsButton';
 
 const OP_FILTER_GROUP_HEADER = 'Op';
 const MAX_EVAL_COMPARISONS = 5;
-const MAX_BULK_DELETE = 10;
+const MAX_SELECT = 100;
```
Comment (gtarpenning) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTable.tsx:85]: we support up to 100 in the backend, and most people who delete more than 5 calls delete 10, presumably they are hitting this max. Easy to revert if this causes problems.

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CompareEvaluationsPage/CompareEvaluationsPage.tsx [hunk 2]
```diff
@@ -73,7 +73,7 @@ export const CompareEvaluationsPage: React.FC<
 
   return (
     <SimplePageLayout
-      title={'Compare Evaluations'}
+      title="Compare Evaluations"
```
Comment (gtarpenning) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CompareEvaluationsPage/CompareEvaluationsPage.tsx:76]: lint

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooksEvaluationComparison.ts [hunk 3]
```diff
@@ -183,8 +183,7 @@ const fetchEvaluationComparisonData = async (
       call.id,
       {
         callId: call.id,
-        // TODO: Get user-defined name for the evaluation
-        name: 'Evaluation',
+        name: call.display_name ?? 'Evaluation',
```
Comment (gtarpenning) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooksEvaluationComparison.ts:186]: allow custom display names in eval page

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooksEvaluationComparison.ts [hunk 4]
```diff
@@ -230,7 +229,11 @@ const fetchEvaluationComparisonData = async (
     // Add the user-defined scores
     evalObj.scorerRefs.forEach(scorerRef => {
       const scorerKey = getScoreKeyNameFromScorerRef(scorerRef);
-      const score = evaluationCallCache[evalCall.callId].output[scorerKey];
+      const output = evaluationCallCache[evalCall.callId].output;
+      if (output == null) {
+        return;
+      }
```
Comment (gtarpenning) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooksEvaluationComparison.ts:235]: return early if the run is failed or running and doesn't have an output

**Review by tssweeney — APPROVED** [review]
Nice

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CompareEvaluationsPage/CompareEvaluationsPage.tsx [hunk 5]
```diff
@@ -165,6 +168,7 @@ const CompareEvaluationsPageInner: React.FC = props => {
           alignItems: 'flex-start',
           gridGap: STANDARD_PADDING * 2,
         }}>
+        <InvalidEvaluationBanner state={state} />
```
Comment (jamie-rasmussen) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CompareEvaluationsPage/CompareEvaluationsPage.tsx:171]: Recommend we don't use "state" as a prop name - potential for confusion

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CompareEvaluationsPage/CompareEvaluationsPage.tsx [hunk 6]
```diff
@@ -237,3 +241,49 @@ const ResultExplorer: React.FC<{state: EvaluationComparisonState}> = ({
...
 };
+
+const InvalidEvaluationBanner: React.FC<{state: EvaluationComparisonState}> = ({
+  state,
+}) => {
+  const [dismissed, setDismissed] = useState(false);
+  const invalidEvals = useMemo(() => {
+    return Object.values(state.data.evaluationCalls)
+      .filter(evalCall => {
+        return Object.keys(evalCall.summaryMetrics).length === 0;
```
Comment (jamie-rasmussen) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CompareEvaluationsPage/CompareEvaluationsPage.tsx:252]: For readability, maybe define this as a short `isValidEval` helper function with a comment about why summaryMetrics indicates that?

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CompareEvaluationsPage/CompareEvaluationsPage.tsx [hunk 7]
```diff
@@ -237,3 +241,49 @@ const ResultExplorer: React.FC<{state: EvaluationComparisonState}> = ({
...
+  return (
+    <Box
+      sx={{
+        width: '100%',
+        paddingLeft: STANDARD_PADDING,
+        paddingRight: STANDARD_PADDING,
+      }}>
+      <Alert
+        severity="info"
+        style={{backgroundColor: hexToRGB(TEAL_300, 0.3), color: '#038194'}}
```
Comment (jamie-rasmussen) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CompareEvaluationsPage/CompareEvaluationsPage.tsx:272]: We shouldn't be using hexcodes for colors, I think this should be teal-600. As a nit there's an effort in general to move towards using tailwind for this sort of styling.

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CompareEvaluationsPage/CompareEvaluationsPage.tsx [hunk 8]
```diff
@@ -237,3 +241,49 @@ const ResultExplorer: React.FC<{state: EvaluationComparisonState}> = ({
...
+            // override the default tailwind classes for text and background hover
+            className="text-override hover:bg-override"
+            variant="ghost"
+            onClick={() => setDismissed(true)}>
+            Dismiss
+          </Button>
+        }>
+        <span style={{fontWeight: 'bold'}}>
+          No summary information found for evaluation
+          {invalidEvals.length > 1 ? 's' : ''}: {invalidEvals.join(', ')}
```
Comment (jamie-rasmussen) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CompareEvaluationsPage/CompareEvaluationsPage.tsx:284]: We have a `maybePluralizeWord` that we should be using (not a big deal but may help if we ever want to localize)

**Review by jamie-rasmussen — APPROVED** [review]

**Pull request merged**